bn_mp_n_root_ex.c

```diff
@@ -89,14 +129,39 @@ int mp_n_root_ex(const mp_int *a, mp_digit b, mp_int *c, int fast)
       if ((res = mp_sub(&t1, &t3, &t2)) != MP_OKAY) {
          goto LBL_T3;
       }
+      /* 
+          Number of rounds is at most log_2(root). If it is more it
+          got stuck, so break out of the loop and do the rest manually.
+       */
+      if (ilog2-- == 0) {
+         break;
+      }
    }  while (mp_cmp(&t1, &t2) != MP_EQ);
 
    /* result can be off by a few so check */
+   /* Overshoot by one if root is smaller */
+   for (;;) {
```

Comment:
why is this `for` loop there? AFAIU this will never repeat!?

Reply:
> AFAIU this will never repeat!?

Neither do I think it will, but it is just a conjecture and should not be trusted until proven.
A slightly different case was my first try of adjusting the seed, although with Halley's algorithm instead of Newton's, where I actually measured the errors. I can't find the raw data now but I asked cs.stackexchange including a plot of that data at https://cs.stackexchange.com/questions/56848/numerical-stability-of-halleys-recurrence-for-integer-n-mathrmth-root Yepp, to no avail ;-)

Oh, I just recognised I missed an "s" in the comment.  It should have said
"[The loop underneath ]Overshoot**s** by one if root is smaller"
Will make it a bit clearer if you insist.

Reply:
I should've gone to bed already long ago...